### PR TITLE
Gnuplot compression

### PR DESCRIPTION
**Pull request by drwells**
I recently generated a lot of gnuplot output for creating nice SVG grids. Some of the output files were quite large and I implemented a compression scheme that cuts down the file size by about 50% for curved grids by only using one line segment for straight faces (instead of the user provided subdivision, which is really intended for only curved faces).

This will probably end up breaking a ton of tests. I'm putting it up as a PR now just so I do not forget about it; I'll finish it up when 9.0 is done.

### CODE REVIEW

File: source/grid/grid_out.cc [hunk 1]
```diff
@@ -3193,8 +3193,45 @@ namespace internal
           q_projector = new Quadrature<dim> (QProjector<dim>::project_to_all_faces(quadrature));
         }
 
+      std::set<typename Triangulation<dim,spacedim>::active_cell_iterator> interior_cells_to_not_skip;
+      std::vector<Point<spacedim>> point_cache;
       for (; cell!=endc; ++cell)
         {
+          if (!cell->at_boundary() &&
+              interior_cells_to_not_skip.find(cell) == interior_cells_to_not_skip.end())
```
Comment (bangerth) [inline source/grid/grid_out.cc]: This last line can now be written slightly more elegantly as
```
  interior_cells_...skip.count(cell) > 0
```

Comment (bangerth) [reply source/grid/grid_out.cc]: actually, I'm not sure whether I just wrote down the correct condition or its opposite, but that points out a readability problem: variable names with negative names are difficult to understand, and even more so if they have a double negative: "not" and "skip". Can you rename it somehow, and maybe document its purpose where you declare it?

File: source/grid/grid_out.cc [hunk 2]
```diff
@@ -3193,8 +3193,45 @@ namespace internal
...
+            {
+              bool skip_cell = true;
```
Comment (bangerth) [inline source/grid/grid_out.cc]: can you explain here what the following lines do, and why?

File: source/grid/grid_out.cc [hunk 3]
```diff
@@ -3935,9 +3978,7 @@ namespace internal
       for (LineList::const_iterator line=line_list.begin();
            line!=line_list.end(); ++line)
         if (eps_flags_base.color_lines_level && (line->level > 0))
-          // lines colored according to
-          // refinement level,
-          // contributed by J�rg
+          // lines colored according to refinement level, contributed by Jörg
```
Comment (bangerth) [inline source/grid/grid_out.cc]: we don't usually annotate with the author here -- just remove it

**Comment by drwells** [comment]
Closed in favor of #6632 and a future PR that will implement the cell-dropping optimization.

**Pull request closed without merging**